### PR TITLE
Enable access to CMV12K SPI registers

### PR DESCRIPTION
**Pull request by tpwrules**
This PR adds all the necessary infrastructure to be able to read and write CMV12K SPI registers through a Linux SPI device and the nMigen gateware.

### CODE REVIEW

File: applets/cmv12k/spi_test.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    def elaborate(self, platform: BetaPlatform):
+        m = Module()
+
+        platform.ps7.fck_domain(requested_frequency=100e6)
+
+        sensor = platform.request("sensor")
+        platform.ps7.fck_domain(250e6, "sensor_clk")
+        m.d.comb += sensor.lvds_clk.eq(ClockSignal("sensor_clk"))
+        m.d.comb += sensor.reset.eq(~self.sensor_reset_n)
```
Comment (anuejn) [inline applets/cmv12k/spi_test.py:18]: shouldnt the inversion be rather handled by the platform?

File: naps/cores/peripherals/bitbang_spi.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        self.devicetree_name = "bitbang_spi" + name_suffix
+        self.pins = pins
+
+    def elaborate(self, platform: SocPlatform):
+        m = Module()
+
+        # for some reason, despite CS being specified as active high in the
+        # device tree, linux and/or the python spidev module treat it as active
+        # low. thus we invert CS here to make the pin active high as defined
+        pins = Record.like(self.pins)
```
Comment (anuejn) [inline naps/cores/peripherals/bitbang_spi.py:21]: why add that level of indirection to all pins?

File: naps/cores/peripherals/mmio_gpio.py [hunk 3]
```diff
@@ -36,8 +36,8 @@ def elaborate(self, platform):
         devicetree_overlay(platform, self.devicetree_name, overlay_content, {"set": self.set, "dat": self.dat, "dirout": self.dirout})
 
         for i, pad in enumerate(self._pads):
-            m.d.comb += self.dat[i].eq(pad.i)
-            m.d.comb += pad.oe.eq(self.dirout[i])
-            m.d.comb += pad.o.eq(self.set[i])
+            if hasattr(pad, "i"): m.d.comb += self.dat[i].eq(pad.i)
```
Comment (anuejn) [inline naps/cores/peripherals/mmio_gpio.py:39]: nit: could you reformat this to not use single line if

**Review by anuejn — CHANGES_REQUESTED** [review]

File: applets/cmv12k/spi_test.py (hunk 1 above)
Comment (tpwrules) [reply applets/cmv12k/spi_test.py:18]: i converted this to non-inverted logic, unlike the other applets, but it makes more sense that way

File: naps/cores/peripherals/bitbang_spi.py (hunk 2 above)
Comment (tpwrules) [reply naps/cores/peripherals/bitbang_spi.py:21]: good point, fixed

File: naps/cores/peripherals/mmio_gpio.py (hunk 3 above)
Comment (tpwrules) [reply naps/cores/peripherals/mmio_gpio.py:39]: done

**Pull request merged**